### DISCUSSION TITLE
Clickhouse replica cannot reach other replicas and crashes with an DNSResolver error

### DISCUSSION THREAD

**Discussion by Programmeris** [ClickHouse/ClickHouse#52686] [Q&A]
I have a Clickhouse cluster consisting of 1 shard and 3 replicas. Everything is deployed to Kubernetes using Altinity Clickhouse Operator. Docker image: clickhouse/clickhouse-server:23.3.8.21-alpine.

Situation: ALL three replicas crash due to various reasons (node ​​failure or HDD/SSD failure). After some time, one of the replicas tries to go up (the other two are also unavailable at this moment) and crashes with a DNSResolver error(All Zookeeper nodes are available at this moment).

Question: can I achieve a behavior in which one replica will be raised ignoring the unavailability of others? 

In the documentation, I found the skip_unavailable_shards and dns_max_consecutive_failures parameters. Maybe a combination of these parameters will solve my case?

**Comment by UnamedRus** [comment]
> crashes 

Do you mean segfault and real crash or something else just shutdown?

**Comment by Programmeris** [comment]
My case:
- Failed HDD RAID
- All three Clickhouse replicas are out of order
- After replacing the failed HDD, ONE Clickhouse replica became available, which went into restart with an error:
`<Error> DNSResolver: cannot resolve host(<replica host>)`
